### PR TITLE
9 - split code

### PR DESCRIPTION
**Pull request by pniedzwiedzinski**
As I commented in #9 the CSS could be split. I've done it on a few components and I want to know, what you think about it.

I also added storybook since it's easier than creating temp vue app for testing.

### CODE REVIEW

File: src/components/BoringHamburger.vue [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<template>
+  <plain-hamburger class="hamburger--boring" @toggle="$emit('toggle')" />
```
Comment (imfaber) [inline src/components/BoringHamburger.vue]: I would automatically pass any event to the parent with `v-on="$listeners"`
```suggestion
  <plain-hamburger class="hamburger--boring" v-on="$listeners />
```

PS. the same for the other components

File: package.json [hunk 2]
```diff
@@ -24,9 +24,11 @@
     "test": "npm run unit",
     "start": "cross-env NODE_ENV=development webpack-dev-server --open --hot",
     "build-docs": "npm run clean-docs && cross-env NODE_ENV=production webpack --progress --hide-modules",
-    "clean-docs": "rimraf docs/dist"
+    "clean-docs": "rimraf docs/dist",
+    "storybook": "start-storybook --port 6006"
   },
   "devDependencies": {
+    "@storybook/vue": "^5.2.8",
```
Comment (imfaber) [inline package.json:31]: ❤️

**Review by imfaber — CHANGES_REQUESTED** [review]
Hi @pniedzwiedzinski 

Great job! I love your proposal.
My only concern is about having a tag per component. 
I'd rather have one tag where I can specify the hamburger type `<vue-tasty-burger type="slider">`.

This could be achieved with dynamic components.
Here a quick example.
```js
<template>
  <component v-bind:is="components[type]" v-on="$listeners"></component>
</template>

<script>
const components = {
  slider: () => import('./SliderHamburger.vue')
  // ... the other components
}

export default {
  components,
  props: {
    type: String
  },
  data() {
    return {
      components
    }
  }
}
</script>
```

Does it make sense?
Any thought?

Many thanks for your work.

**Comment by pniedzwiedzinski** [comment]
I wasn't familiar with dynamic components, and this preserves more compatibility with previous versions, which is great.
Ok, I will keep working on that

**Comment by pniedzwiedzinski** [comment]
I have problems with rollup and I couldn't make it work. Maybe you could check it and see what's wrong?

**Comment by imfaber** [comment]
Sorry, I've been quite busy lately.. I'll have a look ASAP

**Pull request merged**